TERRA-55 ⁃ Create DB test failing on region Set(k, v)

/usr/local/go/bin/go test -c -o /tmp/___1TestDatabase_in_github_com_datastax_terraform_provider_astra_v2_internal_provider.test -gcflags all=-N -l github.com/datastax/terraform-provider-astra/v2/internal/provider #gosetup
/usr/local/go/bin/go tool test2json -t /home/tato/.local/share/JetBrains/IntelliJIdea2021.1/go/lib/dlv/linux/dlv --listen=0.0.0.0:40559 --headless=true --api-version=2 --check-go-version=false --only-same-user=false exec /tmp/___1TestDatabase_in_github_com_datastax_terraform_provider_astra_v2_internal_provider.test -- -test.v -test.paniconexit0 -test.run ^\QTestDatabase\E$
API server listening at: [::]:40559
=== RUN   TestDatabase
2022/04/05 15:50:13 [DEBUG] GET https://api.astra.datastax.com/v2/regions/serverless
2022/04/05 15:50:14 [DEBUG] POST https://api.astra.datastax.com/v2/databases
2022/04/05 15:50:16 [DEBUG] GET https://api.astra.datastax.com/v2/databases/246e3785-6744-4c9d-bbb8-1ff20e612fcc
2022/04/05 15:50:16 [DEBUG] GET https://api.astra.datastax.com/v2/databases/246e3785-6744-4c9d-bbb8-1ff20e612fcc
2022/04/05 15:50:17 [DEBUG] GET https://api.astra.datastax.com/v2/databases/246e3785-6744-4c9d-bbb8-1ff20e612fcc
2022/04/05 15:50:19 [DEBUG] GET https://api.astra.datastax.com/v2/databases/246e3785-6744-4c9d-bbb8-1ff20e612fcc
2022/04/05 15:50:23 [DEBUG] GET https://api.astra.datastax.com/v2/databases/246e3785-6744-4c9d-bbb8-1ff20e612fcc
2022/04/05 15:50:31 [DEBUG] GET https://api.astra.datastax.com/v2/databases/246e3785-6744-4c9d-bbb8-1ff20e612fcc
2022/04/05 15:50:42 [DEBUG] GET https://api.astra.datastax.com/v2/databases/246e3785-6744-4c9d-bbb8-1ff20e612fcc
2022/04/05 15:50:52 [DEBUG] GET https://api.astra.datastax.com/v2/databases/246e3785-6744-4c9d-bbb8-1ff20e612fcc
2022/04/05 15:51:02 [DEBUG] GET https://api.astra.datastax.com/v2/databases/246e3785-6744-4c9d-bbb8-1ff20e612fcc
2022/04/05 15:51:12 [DEBUG] GET https://api.astra.datastax.com/v2/databases/246e3785-6744-4c9d-bbb8-1ff20e612fcc
2022/04/05 15:51:22 [DEBUG] GET https://api.astra.datastax.com/v2/databases/246e3785-6744-4c9d-bbb8-1ff20e612fcc
2022/04/05 15:51:32 [DEBUG] GET https://api.astra.datastax.com/v2/databases/246e3785-6744-4c9d-bbb8-1ff20e612fcc
2022/04/05 15:51:42 [DEBUG] GET https://api.astra.datastax.com/v2/databases/246e3785-6744-4c9d-bbb8-1ff20e612fcc
2022/04/05 15:51:52 [DEBUG] GET https://api.astra.datastax.com/v2/databases/246e3785-6744-4c9d-bbb8-1ff20e612fcc
panic: reflect: call of reflect.Value.IsNil on array Value

goroutine 325 [running]:
reflect.Value.IsNil(0x662fd80, 0xc00168bae0, 0x91, 0x64dfd00)
	/usr/local/go/src/reflect/value.go:1086 +0x1b7
github.com/mitchellh/mapstructure.(*Decoder).decodeSlice(0xc001554ae0, 0x0, 0x0, 0x662fd80, 0xc00168bae0, 0x64dfd40, 0xc00089fc98, 0x197, 0x0, 0x0)
	/home/tato/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.3/mapstructure.go:1091 +0x8b2
github.com/mitchellh/mapstructure.(*Decoder).decode(0xc001554ae0, 0x0, 0x0, 0x662fd80, 0xc00168bae0, 0x64dfd40, 0xc00089fc98, 0x197, 0x0, 0x0)
	/home/tato/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.3/mapstructure.go:469 +0xe25
github.com/mitchellh/mapstructure.(*Decoder).Decode(0xc001554ae0, 0x662fd80, 0xc00168bae0, 0x0, 0x0)
	/home/tato/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.3/mapstructure.go:398 +0x108
github.com/mitchellh/mapstructure.Decode(0x662fd80, 0xc00168bae0, 0x6465740, 0xc00089fc98, 0x0, 0x0)
	/home/tato/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.3/mapstructure.go:302 +0x185
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).setList(0xc00089f470, 0xc00168bb90, 0x1, 0x1, 0x662fd80, 0xc00168bae0, 0x0, 0x0)
	/home/tato/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/field_writer_map.go:126 +0x1ba
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).set(0xc00089f470, 0xc00168bb90, 0x1, 0x1, 0x662fd80, 0xc00168bae0, 0x0, 0x0)
	/home/tato/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/field_writer_map.go:103 +0x3bb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).WriteField(0xc00089f470, 0xc00168bb90, 0x1, 0x1, 0x662fd80, 0xc00168bae0, 0x0, 0x0)
	/home/tato/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/field_writer_map.go:89 +0x80b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc00206c500, 0x7cf3223, 0x7, 0x662fd80, 0xc00168bae0, 0x0, 0x0)
	/home/tato/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/resource_data.go:227 +0x20f
github.com/datastax/terraform-provider-astra/v2/internal/provider.setDatabaseResourceData(0xc00206c500, 0xc0021f2b60, 0x0, 0x0)
	/home/tato/go/src/github.com/datastax/terraform-provider-astra/internal/provider/resource_database.go:489 +0x1c9
github.com/datastax/terraform-provider-astra/v2/internal/provider.waitForDatabaseAndUpdateResource.func1(0x0)
	/home/tato/go/src/github.com/datastax/terraform-provider-astra/internal/provider/resource_database.go:469 +0x9bf
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.RetryContext.func1(0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/tato/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/resource/wait.go:27 +0x7b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext.func1(0xc0016ac840, 0xc00032e690, 0xc000fdef00, 0xc0006d70c0, 0xc0006d8bc0, 0xc0006d8bb8)
	/home/tato/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/resource/state.go:110 +0x2f6
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext
	/home/tato/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/resource/state.go:83 +0x2f6

Debugger finished with the exit code 0



┆Issue is synchronized with this [Jira Task](https://datastax-oss.atlassian.net/browse/TERRA-55) by [Unito](https://www.unito.io)
┆friendlyId: TERRA-55
┆priority: Major

--- a/internal/provider/resource_database.go
+++ b/internal/provider/resource_database.go
@@ -505,7 +505,7 @@ func flattenDatabase(db *astra.Database) map[string]interface{} {
 		"data_endpoint_url":    astra.StringValue(db.DataEndpointUrl),
 		"cqlsh_url":            astra.StringValue(db.CqlshUrl),
 		"cloud_provider":       "",
-		"regions":              [1]string{astra.StringValue(db.Info.Region)},
+		"regions":              []string{astra.StringValue(db.Info.Region)},
 		"keyspace":             astra.StringValue(db.Info.Keyspace),
 		"additional_keyspaces": astra.StringSlice(db.Info.AdditionalKeyspaces),
 		"node_count":           db.Storage.NodeCount,

